ethmonitor/ethmonitor.go

```diff
@@ -352,7 +351,7 @@ func (m *Monitor) listenNewHead() <-chan uint64 {
 					streamingErrorLastTime = time.Now()
 					goto reconnect
 
-				case <-blockTimer.C:
+				case <-time.After(3 * m.options.ExpectedBlockInterval):
```

Comment:
this would create an extra channel + timeout interrupt in each `for loop` iteration

I think let's stick to blockTimer -- and stop it on error / ctx done